plutus-chain-index-core/src/Plutus/ChainIndex/Tx.hs

```diff
@@ -78,7 +79,7 @@ fromOnChainTx = \case
                 let (validatorHashes, otherDataHashes, redeemers) = validators txInputs in
                 ChainIndexTx
                     { _citxTxId = txId tx
-                    , _citxInputs = txInputs
+                    , _citxInputs = Set.toList txInputs
```

Comment:
This is incorrect but to fix it we need to fix the `Tx` from `plutus-core`. Or create our own `Tx` with `[TxIn]` instead of `Set TxIn`.

Reply:
Can you add a TODO and a story for this? I guess it will need to be done once `next-node` is merged in `main`.

Reply:
No need imho, this is good enough.